src/test/java/org/minions/devfund/reinaldo/WhackAMoleTest.java

```diff
@@ -0,0 +1,170 @@
+package org.minions.devfund.reinaldo;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * It is in charged to define the unit tests for WhackAMole class.
+ */
+public class WhackAMoleTest {
+    public static final int NUM_ATTEMPTS =  10;
+    public static final int GRID_DIMENSION = 10;
+    private WhackAMole whackAMole;
+
+    /**
+     *Before tests setup.
+     */
+    @Before
+    public void setUp() {
+        whackAMole = new WhackAMole(NUM_ATTEMPTS, GRID_DIMENSION);
+    }
+
+    /**
+     * This test get the wack a moles in the game.
+     *
+     */
+    @Test
+    public void testWhenIsFillingTheMoleGridWithDefaultCharacter() {
+        whackAMole.createTheMoleGrid();
+        char[][] actualResult = whackAMole.getMoleGrid();
+        char expectedResult = '*';
+
+        assertEquals(expectedResult, actualResult[0][0]);
+        assertEquals(expectedResult, actualResult[1][1]);
+        assertEquals(expectedResult, actualResult[2][2]);
+        assertEquals(expectedResult, actualResult[GRID_DIMENSION - 1][GRID_DIMENSION - 1]);
+    }
+
+    /**
+     * Place mole in row and column that is already taken test.
+     */
+    @Test
+    public void testWhenThePlaceMoleInPlaceWithCharacterDefault() {
+        final int postX = 1;
+        final int postY = 1;
+
+        whackAMole.createTheMoleGrid();
+        whackAMole.place(postX, postY);
+
+        assertFalse(whackAMole.place(postX, postY));
+    }
+
+    /**
+     * This test for update the mole .
+     *
+     */
+    @Test
+    public void testForiIsUpdateTheMoleGrid() {
+        final int postX = 1;
+        final int postY = 1;
+
+        whackAMole.createTheMoleGrid();
+        assertTrue(whackAMole.place(postX, postY));
+    }
+
+    /**
+     * update the Place mole in row and column that is already taken test.
+     */
+    @Test
+    public void testUpdateThePlaceMoleWithCharacterM() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Update this method so that its implementation is not identical to "testForiIsUpdateTheMoleGrid" on line 61. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4144)